cranelift/codegen/src/machinst/lower.rs

```diff
@@ -583,27 +611,48 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
         // end of the BB (in the toplevel driver `lower()`).
         for inst in self.f.layout.block_insts(block).rev() {
             let data = &self.f.dfg[inst];
+            let has_side_effect = has_lowering_side_effect(self.f, inst);
+            // If side-effectful inst has been sunk to another location, skip
+            // it.
+            let inst_sunk = has_side_effect && self.inst_sunk.contains(&inst);
```

Comment:
nitpick: it seems worth pulling this out into an `is_inst_sunk` helper method to clarify the meaning of this code without being forced to look at the details of the data structures and representation

Reply:
Done, thanks!